From 248efffbe8907d5a80c6c8debf2bc2c5896e5bf0 Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Wed, 12 Jul 2017 17:31:48 +0800
Subject: [PATCH 2113/5242] MLK-15959: ASoc: fsl: fix hifi4 driver build
 warning

commit  372cf7fcfc4e23b76c568553ef457ecf4625151e from
https://source.codeaurora.org/external/imx/linux-imx.git

When building sound/soc/fsl/fsl_hifi4.c file, a warning
occurs:
      warning: cast to pointer from integer of different
      size [-Wint-to-pointer-cast]
      (struct timestamp_info_t *)pext_msg->dtstamp;

By forced conversing int type to long type to fix this
issue.

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_hifi4.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_hifi4.c b/sound/soc/fsl/fsl_hifi4.c
index bf8aa65..19c5d5e 100644
--- a/sound/soc/fsl/fsl_hifi4.c
+++ b/sound/soc/fsl/fsl_hifi4.c
@@ -1007,7 +1007,7 @@ int process_act_complete(struct fsl_hifi4 *hifi4_priv, u32 msg)
 			codec_iobuf_info->proc_id = pext_msg->proc_id;
 			hifi4_priv->is_done = 1;
 			hifi4_priv->dpu_tstamp =
-			    (struct timestamp_info_t *)pext_msg->dtstamp;
+			  (struct timestamp_info_t *)((long)pext_msg->dtstamp);
 			hifi4_priv->ret_status = pext_msg->ret;
 			complete(&hifi4_priv->cmd_complete);
 		}
-- 
1.7.9.5

